packages/xo-web/src/xo-app/settings/remotes/remote.js

```diff
@@ -56,12 +56,18 @@ export default decorate([
           name,
           options = remote.options || '',
           password = remote.password,
-          path = remote.path,
           port = remote.port,
           proxyId = remote.proxy,
           type = remote.type,
           username = remote.username,
+          parsedPath,
+          bucket = parsedPath != null && parsedPath.split('/')[0],
+          directory = parsedPath != null && parsedPath.split('/')[1],
         } = state
+        let { path = remote.path } = state
+        if (type === 's3') {
+          path = bucket + '/' + directory
```

Comment:
How will that work if `bucket` and or `directory` is null?

Reply:
neither bucket nor directory should be null, for now there should be no way to do that.

Reply:
From what I read above, I feel like `bucket` and `directory` could be `false` if `parsedPath` is `null`.

If that cannot happen when `type === 's3'`, maybe the destructuring of these values should be moved in the `if`.